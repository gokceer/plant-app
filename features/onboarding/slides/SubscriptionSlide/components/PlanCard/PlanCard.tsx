import { RadioButton } from "@/components";
import { Pressable, StyleSheet, Text, View } from "react-native";

const PlanCard = ({
  title,
  description,
  selected,
  badge,
  testID,
  onPress,
}: {
  title: string;
  description: string;
  selected: boolean;
  badge?: string;
  testID: string;
  onPress: () => void;
}) => (
  <Pressable
    onPress={onPress}
    accessibilityRole="radio"
    accessibilityState={{ selected }}
    style={[styles.planBox, selected ? styles.activePlan : styles.inactivePlan]}
    testID={testID}
  >
    <View style={styles.radioRow}>
      <RadioButton selected={selected} />
      <View style={{ flex: 1 }}>
        <View style={styles.planHeader}>
          <Text style={styles.planTitle}>{title}</Text>
          {badge && (
            <View style={styles.badge}>
              <Text style={styles.badgeText}>{badge}</Text>
            </View>
          )}
        </View>
        <Text style={styles.planText}>{description}</Text>
      </View>
    </View>
  </Pressable>
);

export default PlanCard;

const styles = StyleSheet.create({
  planBox: {
    padding: 16,
    borderRadius: 14,
    marginBottom: 16,
  },
  inactivePlan: {
    backgroundColor: "#FFFFFF0D",
  },
  activePlan: {
    borderColor: "#28AF6E",
    borderWidth: 1.5,
  },
  planTitle: {
    color: "#fff",
    fontSize: 16,
    fontWeight: "bold",
    fontFamily: "RubikMedium",
  },
  planText: {
    color: "#FFFFFFB2",
    fontSize: 14,
    marginTop: 4,
    fontFamily: "RubikLight",
  },
  planHeader: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    position: "relative",
  },
  badge: {
    backgroundColor: "#34C759",
    paddingHorizontal: 12,
    paddingVertical: 4,
    position: "absolute",
    right: -16,
    top: -16,
    borderTopRightRadius: 12,
    borderBottomLeftRadius: 20,
  },
  badgeText: {
    color: "#fff",
    fontSize: 12,
    fontWeight: "bold",
    fontFamily: "RubikMedium",
  },
  radioRow: {
    flexDirection: "row",
    alignItems: "center",
    gap: 12,
  },
});
