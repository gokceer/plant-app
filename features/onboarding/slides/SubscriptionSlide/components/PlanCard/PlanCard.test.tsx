import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import PlanCard from "./PlanCard";

describe("PlanCard", () => {
  const defaultProps = {
    title: "1 Year",
    description: "First 3 days free, then $529.99/year",
    selected: false,
    onPress: jest.fn(),
    testID: "planCard1year",
  };

  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<PlanCard {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders badge when provided", () => {
    const props = { ...defaultProps, badge: "Save 50%" };
    const { getByText } = render(<PlanCard {...props} />);

    expect(getByText("Save 50%")).toBeTruthy();
  });

  it("does not render badge when not provided", () => {
    const { queryByText } = render(<PlanCard {...defaultProps} />);

    expect(queryByText("Save 50%")).toBeNull();
  });

  it("calls onPress when pressed", () => {
    const { getByTestId } = render(<PlanCard {...defaultProps} />);
    const pressable = getByTestId(defaultProps.testID);

    fireEvent.press(pressable);
    expect(defaultProps.onPress).toHaveBeenCalledTimes(1);
  });
});
