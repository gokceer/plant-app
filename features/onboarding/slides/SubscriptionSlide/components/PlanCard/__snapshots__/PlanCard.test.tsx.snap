// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanCard renders correctly and matches snapshot 1`] = `
<View
  accessibilityRole="radio"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 14,
        "marginBottom": 16,
        "padding": 16,
      },
      {
        "backgroundColor": "#FFFFFF0D",
      },
    ]
  }
  testID="planCard1year"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF26",
            "borderRadius": 12,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
          false,
        ]
      }
      testID="radioOuterView"
    />
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "position": "relative",
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "RubikMedium",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          1 Year
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#FFFFFFB2",
            "fontFamily": "RubikLight",
            "fontSize": 14,
            "marginTop": 4,
          }
        }
      >
        First 3 days free, then $529.99/year
      </Text>
    </View>
  </View>
</View>
`;
